### PR TITLE
create profile page

### PR DESCRIPTION
**Pull request by Ephrem758**
I created a profile page. I modularized it into smaller parts which went in the widget folder ,it contains all the reusable widgets that I used in my main file. I also implemented the sizes by using media query so that it can be responsive.  

profile_page

### CODE REVIEW

File: Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+                  ),
+                ],
+              ),
+            ),
+            Row(
+              children: [
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(30, 10, 10, 0),
+                  child: Container(
+                    width: 100,
```
Comment (Redi22) [inline Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart:50]: Avoid using fixed sizing. Make sure you replace it with media query in all the files.

File: Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+                      child: Text(
+                        'BIG DATA                                  ',
+                        style: TextStyle(
+                            fontSize: 17, color: Colors.blue.withOpacity(0.6)),
+                      ),
+                    ),
+                    const Padding(
+                      padding: EdgeInsets.fromLTRB(0, 0, 0, 20),
+                      child: Text(
+                        'Why Big Data Needs Thick Data?',
```
Comment (Redi22) [inline Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart:75]: Declare a variable with the text as a value and use that in the Text widget. it is cleaner and better for later integration as well.

File: Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+//The Second card in my posts section
+
+class Second_post extends StatefulWidget {
```
Comment (Redi22) [inline Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart:118]: First post and Second post are not needed you can build it with a list view and use one widget. you can name the widget PostCard or something else more descriptive

File: Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart
Comment (Redi22) [inline Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart:1]: There are 3 widgets here in the same file which is a bad practice. They should be separate files. One file One class

File: Mobile/starter_project_matador/lib/features/profile page/presentation/widget/profile_description_widget.dart
Comment (Redi22) [inline Mobile/starter_project_matador/lib/features/profile page/presentation/widget/profile_description_widget.dart:1]: you shouldn't add widget to the name here.

**Review by Redi22 — CHANGES_REQUESTED** [review]

File: Mobile/starter_project_matador/lib/features/profile page/presentation/widget/header_widget.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+class Profile_Text extends StatelessWidget {
+  const Profile_Text({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const Padding(
+      padding: EdgeInsets.fromLTRB(30, 15, 0, 0),
+      child: Text(
+        "Profile",
+        style: TextStyle(
```
Comment (BekyH) [inline Mobile/starter_project_matador/lib/features/profile page/presentation/widget/header_widget.dart:12]: Define All the styles in the core/utils directory and reuse them whenever needed.

File: Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+              child: Row(
+                children: [
+                  const Text(
+                    'My Posts',
+                    style: TextStyle(fontSize: 18, fontWeight: FontWeight.w600),
+                  ),
+                  SizedBox(width: screenWidth.size.width * 0.55),
+                  const Icon(
+                    Icons.grid_view,
+                    color: Color.fromARGB(255, 6, 90, 159),
```
Comment (BekyH) [inline Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart:34]: Avoid giving colors in this way. Define All the colors in the core/utils directory and reuse them whenever needed.

**Review by BekyH — CHANGES_REQUESTED** [review]

File: Mobile/starter_project_matador/lib/features/profile page/presentation/widget/header_widget.dart [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/material.dart';
+
+class Profile_Text extends StatelessWidget {
```
[the remark this answers is not in the document]
Comment (Ephrem758) [reply Mobile/starter_project_matador/lib/features/profile page/presentation/widget/header_widget.dart]: Resolved now

File: Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart (hunk 3 above)
Comment (Ephrem758) [reply Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart:118]: Its modularized into one class now.

File: Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart
Comment (Ephrem758) [reply Mobile/starter_project_matador/lib/features/profile page/presentation/widget/posts_widget.dart:1]: resolved

File: Mobile/starter_project_matador/lib/features/profile page/presentation/widget/profile_description_widget.dart
Comment (Ephrem758) [reply Mobile/starter_project_matador/lib/features/profile page/presentation/widget/profile_description_widget.dart:1]: renamed my widget

File: Mobile/starter_project_matador/lib/features/profile page/presentation/screen/profile_page.dart [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                    bottom: -25,
+                    left: screenWidth.size.width * 0.08,
+                    right: screenWidth.size.width * 0.08,
+                    child: StatusCard()),
+              ]),
+
+          //the second half of the page starts here
+          Expanded(
+            child: ListView(
+              physics: NeverScrollableScrollPhysics(),
```
[the remark this answers is not in the document]
Comment (Ephrem758) [reply Mobile/starter_project_matador/lib/features/profile page/presentation/screen/profile_page.dart:51]: Made it scrollable now

**Pull request closed without merging**